src/platform/intel/cavs/lib/clk.c

```diff
@@ -57,12 +57,13 @@ static inline void select_cpu_clock_hw(int freq_idx, bool release_unused)
 		io_reg_write(SHIM_BASE + SHIM_CLKCTL,
 			     (io_reg_read(SHIM_BASE + SHIM_CLKCTL) &
 			      ~SHIM_CLKCTL_OSC_REQUEST_MASK) | enc);
+	}
+
 #if CONFIG_TIGERLAKE
-		/* TGL specific HW recommended flow */
-		if (freq_idx != CPU_HPRO_FREQ_IDX)
-			pm_runtime_put(PM_RUNTIME_DSP, PWRD_BY_HPRO | (CONFIG_CORE_COUNT - 1));
+	/* TGL specific HW recommended flow */
+	if (freq_idx != CPU_HPRO_FREQ_IDX)
```

Comment:
is this supposed to be the opposite condition to the pm get above

Reply:
yes, so this will make the get/put happen in pairs an balance. 

it is not put together with the get above in 'else', as the sequence for put/get is difference:

get (disabling power gating) -> clock switching -> put (enabling power gating).

Reply:
Ah yes this makes sense now, they are set on clock enable/disable. I understand now, clever thanks!